src/charm.py

```diff
@@ -74,12 +75,28 @@ def __init__(self, *args):
         self.cassandra = Cassandra(charm=self)
         logging.getLogger("cassandra").setLevel(logging.CRITICAL)
 
+    def config_okay(self, event: EventBase) -> bool:
+        """Ensure required config values are provided and valid."""
+        if "heap_size" not in self.config:
```

Comment:
With a walrus you wouldn't need line 84.

```suggestion
        if not (heap_size := self.config.get("heap_size"):
```